packages/create-commandkit/src/index.ts

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env node
+console.clear();
+
+import type { ModuleType, PackageManager } from './types';
+
+import { intro, text, select, password, outro } from '@clack/prompts';
+import { commandkit, hints, outroMsg } from './utils';
+import { setup } from './functions/setup';
+import { installDeps } from './functions/installDeps';
+import { copyTemplates } from './functions/copyTemplates';
+
+import path from 'path';
+import colors from 'colors';
+import fs from 'fs-extra';
+
+await intro(`Welcome to ${commandkit}!`);
+
+const dir = path.resolve(
+    process.cwd(),
+    (await text({
+        message: 'Enter a project directory:',
+        placeholder: 'Leave blank for current directory',
+        defaultValue: '.',
+        validate: (value) => {
+            value = path.resolve(process.cwd(), value);
+            let isEmpty;
+
+            try {
+                const contents = fs.readdirSync(value);
+                isEmpty = contents.length === 0;
+            } catch {
+                isEmpty = true;
+            }
+
+            return isEmpty ? undefined : 'Directory is not empty!';
+        },
+    })) as string,
+);
+
+const manager = (await select({
+    message: 'Select a package manager:',
+    options: [
+        { label: 'npm', value: 'npm' },
+        { label: 'pnpm', value: 'pnpm' },
+        { label: 'yarn', value: 'yarn' },
+    ],
+})) as PackageManager;
```

Comment:
Do we need typecasting here? Does type annotation not work in this? If it doesn't then keep this.